controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -822,3 +823,12 @@ func (r *StorageConsumerReconciler) getAvailableServices(storageCluster *ocsv1.S
 
 	return availableServices, nil
 }
+
+func removeStorageClusterAsOwner(obj client.Object) {
+	refs := obj.GetOwnerReferences()
+	if idx := slices.IndexFunc(refs, func(owner metav1.OwnerReference) bool {
+		return owner.Kind == "StorageCluster"
+	}); idx != -1 {
+		obj.SetOwnerReferences(slices.Delete(refs, idx, idx+1))
+	}
+}
```

Comment:
This is only called once, please embed it in the call site

Reply:
done.

Reply:
done.